web/src/shared/hooks/use-annotations-taxons.ts

```diff
@@ -15,11 +15,11 @@ export default function useAnnotationsTaxons(annotationsByPages?: PageInfo[]): M
             for (let page of annotationsByPages) {
                 for (let obj of page.objs) {
                     if (
-                        obj.data.dataAttributes?.[0] &&
-                        obj.data.dataAttributes?.[0].value &&
-                        obj.data.dataAttributes?.[0].type === 'taxonomy'
+                        obj.data?.dataAttributes?.[0] &&
```

Comment:
I think it may be better to not check whole path every time and do somethig like this:

```
for (let obj of page.objs) {
    сonst firstAttribute = obj.data?.dataAttributes?.[0];
    
    if (
        firstAttribute?.value &&
        firstAttribute?.type === 'taxonomy'
    ) {
        taxonIdArr.push(firstAttribute.value);
    }
}
```

Or even this:
```

for (let obj of page.objs) {
    сonst { value, type } = obj.data?.dataAttributes?.[0] || {};
    
    if (value && type === 'taxonomy') {
        taxonIdArr.push(value);
    }
}
```